crates/bevy_core_pipeline/src/tonemapping/tonemapping.wgsl

```diff
@@ -1,9 +1,11 @@
-#import bevy_core_pipeline::fullscreen_vertex_shader
+#define TONEMAPPING_PASS
 
-#import bevy_render::view
+#from bevy_core_pipeline::fullscreen_vertex_shader import FullscreenVertexOutput
+#from bevy_render::view import View
+#from bevy_core_pipeline::tonemapping import tone_mapping, powsafe, screen_space_dither
 
 @group(0) @binding(0)
-var<uniform> view: View;
+var<uniform> view: ::View;
```

Comment:
I dislike that this is required and I strongly suspect the Bevy community will react poorly to this as well. If it is impossible / infeasible to remove this, I would almost consider that a deal breaker as people will react with "yuck" (and "yuck" reactions are very bad, even if they are workable solutions). I personally react this way both because of the syntax and because it forces users to think about symbols "imported into the local namespace" as different from symbols "defined in the local namespace". This is not a common logical model and one that will raise some eyebrows.

I'm pretty sure I understand why this is here: when we import a type, we give it a fully formed name (ex: `struct View` inside `bevy_render::view` becomes something like `bevy_render__view__View` when imported. `::View` then expands to `bevy_render__view__View` so it can match the imported type.

When I put my "user" hat on and think about "how people think about imports" , I'm confused why this is necessary. The "mental model" of imports is that if you pull them in, they become a part of the local namespace.

When I put my "engine developer" hat on, I do see where the problems arise. Namely, when we import code from another file, we can't assume it is safe to bring _everything_ into the local namespace. `bevy_render::view` might contain (and rely on) some internal `helper()` function that collides with the current context's `helper()` function. If we import a `view` function that calls that function, we don't want them to collide. I see a couple of implementation paths here for removing the `::` from `::View`:

### (1) Resolve all names to allow "unmodified" resolution in the final global context

When transforming the source of the imported code to the "fully qualified" code for an imported type, do not "fully qualify" imported types. This ensures the internals still work as expected in the imported context (as their references to `View` would also be "unqualified", while also ensuring the local context doesn't require the qualified type. All non-imported types would still be qualified. We are always compiling for one "final" global context.

* view.wgsl
    ```rust
    #define_import_path bevy_render::view
    fn bar() {
    }

    fn baz() {
    }

    fn foo() {
        bar();
        baz();
    }
    ```
* tonemapping.wgsl
    ```rust
    #define_import_path bevy_pbr::tonemapping
    #from bevy_render::view import bar

    fn tonemap() {
        bar();
    }
    ```
* pbr.wgsl
    ```rust
    #from bevy_pbr::tonemapping import tonemap
    #from bevy_render::view import foo, bar 

    // begin imported from view.wgsl
    // no import prefix because imported into global context
    fn foo() {
        bar();
        bevy_pbr__view__baz();
    }

    // no import prefix because imported into global context
    fn bar() {
    }

    fn bevy_pbr__view__baz() {
    }
    // end imported from view.wgsl


    // begin imported from tonemapping.wgsl
    // no import prefix because imported into global context
    fn tonemap() {
        bar();
    }
    // end imported from tonemapping.wgsl

    fn fragment() {
        foo();
        bar();
        tonemap();
    }
    ```

Some implications:
1. I'm pretty sure this is buildable within the current general naga_oil model 
2. We will need to re-process imports for each "final" shader. We cannot reuse work (suboptimal, especially when we are loading shaders at runtime).
3. We need to propagate import information "down" the dependency tree to ensure each imported module knows about the global import context.

### (2) Always use global names, but find and replace "local" imports

Aka what we're already doing. When referencing an imported type, we will replace it with the fully qualified path. For a given file, if we see `View` anywhere, we replace it with the fully qualified `bevy_render__view__View`. Same goes for `view::View` if `bevy_pbr::view` is in scope.

However the first case `View` is challenging because you cannot "just" use regexes like we currently do (or at least, not easily) to extract and replace imports. However because `View` is "valid" wgsl syntax, in theory we (aka Naga) could parse it and we could operate on the AST to find and replace imports with their fully qualified name.

The second case `view::View` is challenging because it _cannot_ be parsed as valid WGSL, so we probably need a different regex-based preprocessing approach to handle these cases.

Some thoughts:
1. We can process imports ahead of time and reuse them across contexts as every reference is always a fully qualified path.
2. I don't (yet) know if Naga would support the `View` substitution case (ex: can we jump into the process after parsing and modify the AST to "fix" references, or will that fail some validation process?).

From my perspective (2) is the better path. Curious what your thoughts on this problem, these paths, and any other paths I missed.

Reply:
i hear you. i didn't think it was that bad myself (the hint that it's imported can be useful) but if it's a dealbreaker then i'll see what i can do.

1 - this gets pretty hairy with deeper trees that share imports. if you have `a` importing `b1` and `b2`, and both `b1` and `b2` import `foo` from `c` (but `a` doesn't), what name should we use for `foo`? i'll think a bit more but i'm not sure this is the way forwards.

2 - leaving `View` in the importer would fail validation. we need a fully formed valid shader source to get to IR (which is reasonable - the generated IR can vary depending on what sort of thing an identifier refers to).

The reason i required the `::` was because i can't (without an AST context) determine whether the unqualifed identifier `view` is referring to an imported `bevy_pbr::bindings::view` or to a local variable also called `view`. and i can't get to an AST context without a valid shader source... 

but, i think using a regex to capture complete variable names and replace them all will work. this will lead to confusing final output if you shadow an import (your local would get renamed) but since you're strongly opposed to the way it works now, i'll try it.